src/main/java/Duke.java

```diff
@@ -1,10 +1,234 @@
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+//        return "Duke heard: " + input;
+        Parser parse = new Parser(input, false);
+
+        int index;
+        String result = "";
+
+        switch (parse.getCommand()) {
+        case LIST:
+            result += "Here are the tasks in your list: \n";
+            for (int i = 0; i < tasks.size(); i++) {
+                result += (String.format("%d.%s\n", i + 1, tasks.get(i)));
+            }
+
+            return result;
+        case DONE:
+            index = parse.getIndex();
+            tasks.doTask(index);
+
+
+            result += "Nice! I've marked this task as done: \n";
+            result += (" " + tasks.get(index) + "\n");
+            try {
+                Storage.doTask(index);
+            } catch (IOException e) {
+                return e.getMessage();
+            }
+
+            return result;
+        case DELETE:
+            index = parse.getIndex();
+
+            result += "Noted. I've removed this task: \n";
+            result += " " + tasks.remove(index) + "\n";
+
+            try {
+                Storage.deleteTask(index);
+            } catch (IOException e) {
+                return e.getMessage();
+            }
+
+            return result;
+        case FIND:
+            try {
+                String searchTerm = parse.getSearchTerm();
+                result += "Here are the matching tasks in your list:\n";
+                int number = 1;
+                for (int i = 0; i < tasks.size(); i++) {
+                    if (tasks.get(i).toString().contains(searchTerm)) {
+                        result += String.format("%d.%s\n", number++, tasks.get(i));
+                    }
+                }
+
+                return result;
+            } catch (IndexOutOfBoundsException e) {
+                return ("☹ OOPS!!! The search term of a "
+                        + parse.getCommandString() + " cannot be empty.");
+            }
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            try {
+                Task task = parse.createTask();
+                tasks.add(task);
+
+                result += ("Got it. I've added this task: \n");
+                result += (" " + task + "\n");
+                result += (String.format("Now you have %d tasks in the list.\n", tasks.size()));
+
+                try {
+                    Storage.addTask(input);
+                } catch (IOException e) {
+                    return e.getMessage();
+                }
+
+                return result;
+            } catch (IndexOutOfBoundsException e) {
+                return ("☹ OOPS!!! The description of a "
+                        + parse.getCommandString() + " cannot be empty.");
+            }
+        case BYE:
+            return "Press the close button.";
```

Comment:
Should your `main` function be handling the logic for command execution?

Think of how you can abstract the different commands (delete, todo, event, ...) into separate classes and let them handle their own command execution respectively.